src/cpl/nuopc/rof_comp_nuopc.F90

```diff
@@ -471,10 +472,16 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
 
     call ESMF_GridCompGet(gcomp, vm=vm, localPet=localPet, rc=rc)
     if (chkerr(rc,__LINE__,u_FILE_u)) return
-    call ESMF_VMGet(vm, pet=localPet, peCount=nthrds, rc=rc)
+    call ESMF_VMGet(vm, pet=localPet, peCount=localPeCount, rc=rc)
     if (chkerr(rc,__LINE__,u_FILE_u)) return
 
-!$  call omp_set_num_threads(nthrds)
+    if(localPeCount == 1) then
+       call NUOPC_CompAttributeGet(gcomp, "nthreads", value=cvalue, rc=rc)
+       if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=u_FILE_u)) return
+       read(cvalue,*) nthrds
+    else
+       nthrds = localPeCount
+    endif!$  call omp_set_num_threads(nthrds)
```

Comment:
If I understand this correctly, the call to omp_set_num_threads is going to be done regardless of the results of the if branch. I just suggest that this statement should be on its own line to clarify that. As it is it's easy to ignore everything after the endif as normally you can't have anything after that.

This is complex enough that I'd suggest adding a comment above this section to clarify what it's doing.

Reply:
oh - that's a bug, it should be on it's own line.  Correction incoming.